recipe/conda_build_config.yaml

```diff
@@ -0,0 +1,4 @@
+numpy:
+  - 1.18
```

Comment:
This took me forever to work out, but for my own records, the py3.10 parts of this matrix are under `.ci_support/migrations`

Reply:
I'm in a bit of a viral-induced haze this weekend, but maybe if I ask nicely @h-vetinari (SciPy conda-forge maintainer) will offer some input on whether we're on the right track for pinning build-time NumPy appropriately for conda-forge here.

Reply:
> @IAlibay: This took me forever to work out, but for my own records, the py3.10 parts of this matrix are under `.ci_support/migrations`

They generally come from the global [pinning](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/main/recipe/conda_build_config.yaml#L651-L655), but since the py310 migration hasn't been closed yet, it only exists in the migrator so far. 

Looking at the global pinning also shows that (in accordance with NEP29), conda-forge has already uniformly moved to numpy 1.19 as the lowest supported version.

> @tylerjereddy: I'm in a bit of a viral-induced haze this weekend, but maybe if I ask nicely @h-vetinari (SciPy conda-forge maintainer) will offer some input on whether we're on the right track for pinning build-time NumPy appropriately for conda-forge here.

Thanks for the ping. This is one of those things where conda-forge infrastructural considerations & consistency overrides specific projects compatibility settings. In short: I suggest you avoid trying to force back numpy 1.18 support, it's not used in conda-forge anymore.

If `pyproject.toml` gives you troubles during installation, the most reasonable workaround (sadly!) is often to delete it (or patch out the respective sections). conda-forge and pyproject.toml are at odds in several respects, hopefully this situation will improve in the future.

Reply:
Thanks for your thoughts on this @h-vetinari !

I see what you mean re: avoiding forcing numpy 1.18 support, however this is a historical version of MDAnalysis (i.e. already released) where we promised users that the minimum supported version of numpy is 1.18, my understanding is to avoid ABI issues we should be overriding the default 1.19 pin?

re: pyproject.toml it appears that `pip` has reverted the non-optional `pyproject.toml` verification so we can drop the max version pinning I _think_

Reply:
> however this is a historical version of MDAnalysis (i.e. already released) where we promised users that the minimum supported version of numpy is 1.18

I see this merges into the main branch, so it's the most up-to-date available version I guess? It's completely normal that different builds for the same version in conda-forge change dependencies and their lower bounds (that's what most migrators do; the numpy bump just came through rerendering the feedstock though). conda will also take care of making sure that the numpy version is satisfied in the environment.

There wouldn't be any issues for building with 1.18 per se (as the numpy C-API is forward compatible), but it's just not worth the hassle and fragmentation IMO. If you feel it's necessary, that's a choice you can make of course, but as I tried to explain this is one of the points where the promises of the upstream project and the conda-forge packaging can and often do diverge without issue.